src/main/java/moze_intel/projecte/emc/arithmetics/HiddenBigFractionArithmetic.java

```diff
@@ -7,22 +7,12 @@
     @Override
     public BigFraction div(BigFraction a, long b)
     {
-        try
+        BigFraction result = super.div(a, b);
+        if (BigFraction.ZERO.compareTo(result) <= 0 && result.compareTo(BigFraction.ONE) < 0)
         {
-            if (this.isFree(a)) return getFree();
```

Comment:
I'd be surprised if we could skip this line. Otherwise it looks like this could work.

Reply:
We can skip it because it is covered by `super.div(a, b);` Because HiddenBigFractionArithmetic extends FullBigFractionArithmetic. So the isFree check already gets done there.